lib/generators/light_service/organizer_generator.rb

```diff
@@ -0,0 +1,66 @@
+require_relative './generator_utils'
+
+module LightService
+  module Generators
+    class OrganizerGenerator < Rails::Generators::Base
+      include GeneratorUtils
+
+      argument :name, :type => :string
+
+      class_option :dir,
+                   :type => :string,
+                   :default => "organizers",
+                   :desc => "Path to write organizers to"
+
+      class_option :tests,
+                   :type => :boolean,
+                   :default => true,
+                   :desc => "Generate tests (currently only RSpec supported)"
+
+      source_root File.expand_path('templates', __dir__)
+
+      desc <<~DESCRIPTION
+        Description:
+          Will create the boilerplate for an organizer. Pass it an organizer name, e.g.
+            thing_maker, or ThingMaker   - will create ThingMaker in app/organizers/thing_maker.rb
+            thing/maker, or Thing::Maker - will create Thing::Maker in app/organizers/thing/maker.rb
+
+        Options:
+          Skip rspec test creation with --no-tests
+          Write organizers to a specified dir with --dir="workflows". Default is "organizers" in app/organizers
+
+        Full Example:
+          rails g light_service:organizer My::Awesome::Organizer
+      DESCRIPTION
+
+      # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
+      def create_organzier
+        gen_vals = create_required_gen_vals_from(name)
+
+        @module_path     = gen_vals[:module_path]
+        @class_name      = gen_vals[:class_name]
+        @full_class_name = gen_vals[:full_class_name]
+
+        file_name = gen_vals[:file_name]
+        file_path = gen_vals[:file_path]
+
+        root_dir       = options.dir.downcase
+        organizer_dir  = File.join('app', root_dir, *file_path)
+        organizer_file = "#{organizer_dir}/#{file_name}"
+
+        make_nested_dir(organizer_dir)
+        template("organizer_template.erb", organizer_file)
+
+        return unless must_gen_tests?
+
+        spec_dir       = File.join('spec', root_dir, *file_path)
+        spec_file_name = gen_vals[:spec_file_name]
+        spec_file      = "#{spec_dir}/#{spec_file_name}"
+
+        make_nested_dir(spec_dir)
+        template("organizer_spec_template.erb", spec_file)
+      end
```

Comment:
Might be too big of a refactor or even an overkill, but I see an organizer and a couple of actions here. 😉

Reply:
Ha, I looked at the `ActionGenerator` and `OrganizerGenerator` in a split view, it's almost the same code. I see two actions here with an organizer that is reusable between these two generators.

![create-action-and-organizer](https://user-images.githubusercontent.com/80530/87113935-fbfd7080-c234-11ea-95e9-780f5c340137.jpg)

Reply:
You already worked on this PR a ton, I can refactor it after I merged it. No big deal if you don't want to do it.

Reply:
Naah - it's cool, I'll refactor it. It was even more duplicated before I put `create_required_gen_vals` function in. Should be a simple enough effort - will handle it today :) Thanks for the feedback.